pkg/controller/iotconfig/collector.go

```diff
@@ -8,46 +8,24 @@ package iotconfig
 import (
 	"context"
 
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
 	iotv1alpha1 "github.com/enmasseproject/enmasse/pkg/apis/iot/v1alpha1"
 	"github.com/enmasseproject/enmasse/pkg/util/install"
 	appsv1 "k8s.io/api/apps/v1"
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 func (r *ReconcileIoTConfig) processCollector(ctx context.Context, config *iotv1alpha1.IoTConfig) error {
-	return r.processDeployment(ctx, "iot-gc", config, false, r.reconcileCollectorDeployment)
-}
-
-func (r *ReconcileIoTConfig) reconcileCollectorDeployment(config *iotv1alpha1.IoTConfig, deployment *appsv1.Deployment) error {
-
-	install.ApplyDeploymentDefaults(deployment, "iot", deployment.Name)
-
-	err := install.ApplyContainerWithError(deployment, "collector", func(container *corev1.Container) error {
-		if err := install.SetContainerImage(container, "iot-gc", config); err != nil {
-			return err
-		}
-
-		// set default resource limits
 
-		container.Resources = corev1.ResourceRequirements{
-			Limits: corev1.ResourceList{
-				corev1.ResourceMemory: *resource.NewQuantity(128*1024*1024 /* 128Mi */, resource.BinarySI),
-			},
-		}
+	// we no longer need the "iot-gc", so we only remove it
```

Comment:
Why do we need this? To deal with upgrades?

Reply:
Yes. When yo perform an upgrade, it will remove the deployment.